src/devnet-utils.ts

```diff
@@ -56,94 +57,125 @@ export class DevnetUtils implements Devnet {
         return `${this.hre.starknet.networkConfig.url}`;
     }
 
-    private async withErrorHandler<T>(asyncFn: () => Promise<T>, errorMessage: string) {
+    private async requestHandler(
+        url: string,
+        method: Method,
+        errorMessage?: string,
```

Comment:
Instead of passing in the error message, I would suggest constructing it like this just before throwing it:
```typescript
`Request failed: Could not ${method} ${url}. This is a Devnet-specific functionality. Make sure you really want to interact with Devnet and that it is running and available at ${this.endpoint}`
```

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/274/commits/cad87ed19051704fb1e0dc240df16f8583ab878e